ui/src/criteria/classification.tsx

```diff
@@ -440,8 +504,150 @@ function ClassificationEdit(props: ClassificationEditProps) {
   );
 }
 
-function ClassificationInline() {
-  return null;
+type ClassificationInlineProps = {
+  groupId: string;
+  criteriaId: string;
+  data: Data;
+  config: Config;
+};
+
+function ClassificationInline(props: ClassificationInlineProps) {
+  const source = useSource();
+  const classification = source.lookupClassification(
+    props.config.occurrence,
+    props.config.classification
+  );
+  const updateCriteria = useUpdateCriteria(props.groupId, props.criteriaId);
+
+  const hintDataState = useSWRImmutable(
+    {
+      type: "hintData",
+      occurrence: props.config.occurrence,
+      entity: classification.entity,
+      key: props.data.selected[0].key,
+    },
+    async (key) => {
+      const hintData = props.config.valueConfigs
+        ? await source.getAllHintData(key.occurrence, key.entity, key.key)
+        : undefined;
+      return {
+        hintData,
+      };
+    }
+  );
+
+  if (!props.config.valueConfigs) {
+    return null;
+  }
+
+  const onSelect = (event: SelectChangeEvent<string>) => {
+    const {
+      target: { value: sel },
+    } = event;
+    updateCriteria(
+      produce(props.data, (data) => {
+        const config = props.config.valueConfigs?.find(
+          (c) => c.attribute === sel
+        );
+        let attribute = ANY_VALUE;
+        if (config) {
+          attribute = config.attribute;
+        }
+
+        if (attribute === data.valueData.attribute) {
+          return;
+        }
+
+        const newHintData = hintDataState.data?.hintData?.find(
+          (hint) => hint.attribute === attribute
+        );
+
+        data.valueData = {
+          ...data.valueData,
+          attribute,
+          numeric: !!newHintData?.integerHint,
+        };
+
+        if (data.valueData.range.min === 0 && data.valueData.range.max === 0) {
+          data.valueData.range = {
+            id: "",
+            min: newHintData?.integerHint?.min ?? 0,
+            max: newHintData?.integerHint?.max ?? 1000,
```

Comment:
Is this the default integer range when there's no hint data?

Reply:
The user shouldn't be able to get to this point if there's no hint data but the compiler can't know that so I have to provide fallback values.